src/openrct2-ui/windows/Scenery.cpp

```diff
@@ -1019,13 +1089,30 @@ class SceneryWindow final : public Window
         _tabSelections[tabIndex] = value;
     }
 
-    SceneryTabInfo* GetSceneryTabInfoForGroup(const ObjectEntryIndex sceneryGroupIndex)
+    SceneryTabInfo* GetSceneryTabInfoForMisc()
     {
-        if (sceneryGroupIndex == OBJECT_ENTRY_INDEX_NULL)
+        for (auto& tabEntry : _tabEntries)
```

Comment:
misc tab is always the penultimate tab if it exists we don't really need to walk the whole vector. Just check if the penultimate is misc and return it if it is.

Reply:
misc is only the last tab when there is more than one row of tabs, otherwise the all scenery tab would be last. This could be sped up by only looking at the last two tabs but if we ever change this up I don't think it will look as nice.

Reply:
see comment on GetSceneryTabInfoForSearch